### DISCUSSION TITLE
Possible to see args/kwargs for individual function calls?

### DISCUSSION THREAD

**Discussion by TheTechromancer** [bloomberg/memray#666] [Q&A]
Does memray record the args/kwargs of individual function calls?

Basically I just need to be able to tell whether the memory leak was due to lots of small allocations, or one huge one -- and if so, find out what exactly caused that huge allocation.

**Comment by godlygeek** [comment]
No, Memray only records the stack at which each allocation was performed, it doesn't record anything about the local variables or function parameters. It _does_ record the size of each allocation, so you can figure out whether the number of bytes allocated by a given function were spent all in one big allocation or in many small allocations. That's not surfaced well in the `stats` reporter (all you see is a total size and a count for each stack), but the `stats` reporter gives a histogram of allocation sizes, and the with `memray transform csv` you can write out a CSV file containing every allocation that was performed, the allocator used to perform it, the thread the allocation happened in, and the stack trace leading to the allocation.

**Reply by TheTechromancer** [reply]
Okay, thanks for the explanation. I'll have to check out the `transform csv` option, that sounds close to what I want.
